bittensor/protocol.py

```diff
@@ -0,0 +1,348 @@
+        default = "",
+        allow_mutation = True
+    )
+
+    # A unique identifier associated with the axon, set on the axon side.
+    axon_uuid: Optional[ str ] = pydantic.Field(
+        title = 'axon_uuid',
+        description = 'A unique identifier associated with the axon, set on the axon side.',
+        examples = "5ecbd69c-1cec-11ee-b0dc-e29ce36fec1a",
+        default = "",
+        allow_mutation = True
+    )
+
+    # The ss58 encoded hotkey string of the dendrite wallet.
+    dendrite_hotkey: Optional[ str ] = pydantic.Field(
+        title = 'dendrite_hotkey',
+        description = 'The ss58 encoded hotkey string of the dendrites wallet.',
+        examples = "5EnjDGNqqWnuL2HCAdxeEtN2oqtXZw6BMBe936Kfy2PFz1J1",
+        default = "",
+        allow_mutation = True
+    )
+
+    # The bittensor version on the dendrite as an int.
+    axon_hotkey: Optional[ str ] = pydantic.Field(
+        title = 'axon_hotkey',
+        description = 'The ss58 encoded hotkey string of the axon wallet.',
+        examples = "5EnjDGNqqWnuL2HCAdxeEtN2oqtXZw6BMBe936Kfy2PFz1J1",
+        default = "",
+        allow_mutation = True
+    )
+
+    # A signature verifying the tuple (dendrite_nonce, axon_hotkey, dendrite_hotkey, dendrite_uuid)
+    dendrite_signature: Optional[ str ] = pydantic.Field(
+        title = 'dendrite_signature',
+        description = 'A signature verifying the tuple (nonce, axon_hotkey, dendrite_hotkey, dendrite_uuid)',
+        examples = "0x0813029319030129u4120u10841824y0182u091u230912u",
+        default = "",
+        allow_mutation = True
+    )
+    def dendrite_sign(self, keypair):
+        message = f"{self.dendrite_nonce}.{self.dendrite_hotkey}.{self.axon_hotkey}.{self.dendrite_uuid}"
+        self.dendrite_signature = f"0x{keypair.sign(message).hex()}"
+
+    # A signature verifying the tuple (axon_nonce, axon_hotkey, dendrite_hotkey, axon_uuid)
+    axon_signature: Optional[ str ] = pydantic.Field(
+        title = 'axon_signature',
+        description = 'A signature verifying the tuple (axon_nonce, axon_hotkey, dendrite_hotkey, axon_uuid)',
+        examples = "0x0813029319030129u4120u10841824y0182u091u230912u",
+        default = "",
+        allow_mutation = True
+    )
+    def axon_sign(self, keypair):
+        keypair = bittensor.Keypair( ss58_address = self.axon_hotkey )
+        message = f"{self.axon_nonce}.{self.axon_hotkey}.{self.dendrite_hotkey}.{self.axon_uuid}"
+        self.axon_signature = f"0x{keypair.sign(message).hex()}"
+
+class Request( pydantic.BaseModel ):
+
+    class Config:
+        validate_assignment = True
+
+    # A signature verifying the tuple (axon_nonce, axon_hotkey, dendrite_hotkey, axon_uuid)
+    headers: Optional[ Headers ] = pydantic.Field(
+        title = 'headers',
+        description = 'headers associated with the request.',
+        examples = "bt.Headers",
+        default = None,
+        allow_mutation = True,
```

Comment:
By setting `allow_mutation` here it is set for the whole `BaseModel` subclass. No need to add to each `Field()`.  

Even if we wanted to set `allow_mutation = False` here, it seems it wouldn't be enforced after trying myself:
```python
ValueError: On field "dendrite_signature" the following field constraints are set but not enforced: allow_mutation.
```